GitCommands/Git/GitModule.cs

```diff
@@ -2136,10 +2136,19 @@ public string GetEffectiveSetting(string setting)
             return EffectiveConfigFile.GetValue(setting);
         }
 
-        public string GetEffectiveGitSetting(string setting, bool cache = true)
+        public string? GetEffectiveGitSetting(string setting, bool cache = true)
         {
             GitArgumentBuilder args = new("config") { "--includes", "--get", setting };
-            return GitExecutable.GetOutput(args, cache: cache ? GitCommandCache : null).Trim();
+            ExecutionResult result = GitExecutable.Execute(args, cache: cache ? GitCommandCache : null, throwOnErrorExit: false);
+
+            // Handle no value set, is error code 1: https://git-scm.com/docs/git-config#_description
+            const int ConfigKeyInvalidOrNotSet = 1;
+            return result.ExitCode switch
+            {
+                ExecutionResult.Success => result.StandardOutput.Trim(),
+                ConfigKeyInvalidOrNotSet => null,
+                _ => throw new ExternalOperationException("git", args.ToString(), WorkingDir, result.ExitCode, new InvalidOperationException("Error getting config value"))
```

Comment:
We'll need to loc the error message.

Reply:
I do not think it would be worth to add translation to GitCommands.dll. Anyway, this should be are very rare exception.

Reply:
As we're seeing it's not as uncommon :)
But I'm not going to force this, especially, if we want to backport this back to 4.0